libraries/AP_SafeRTL/AP_SafeRTL.h

```diff
@@ -0,0 +1,152 @@
+#pragma once
+
+#include <AP_Buffer/AP_Buffer.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Common/Bitmask.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_AHRS/AP_AHRS.h>
+#include <DataFlash/DataFlash.h>
+#include <GCS_MAVLink/GCS.h>
+
+// definitions and macros
+#define SAFERTL_ACCURACY_DEFAULT        2.0f    // default _ACCURACY parameter value.  Points will be no closer than this distance together.
+#define SAFERTL_POINTS_DEFAULT          150     // default _POINTS parameter value.  High numbers improve path pruning but use more memory and CPU for cleanup. Memory used will be 20bytes * this number.
+#define SAFERTL_POINTS_MAX              500     // the absolute maximum number of points this library can support.
```

Comment:
It's worth adding a static assert to check that SAFERTL_POINTS_MAX is < int16_t max. (Or uint16_t if it gets swapped) otherwise someone on a linux board could easily raise this then be surprised when it doesn't work/they have far less then they expected to.